### ISSUE TITLE
Decision-vs-luck decomposition for hidden-info chance games (banqi/jungle-flip analysis)

### ISSUE THREAD

**Issue by brianhliou** [brianhliou/mistboard#206]
## Problem

Lichess-style analysis grades a move by the eval swing it caused and attributes all of it to the player. That's correct for deterministic games. For **hidden-info chance games** (banqi, jungle-flip), a **flip** move's eval swing has two independent components:

```
eval swing of a flip  =  DECISION quality  +  LUCK
                         (did you pick the     (what the revealed
                          best-EV move?)        tile happened to be)
```

Grading the whole swing blames the player for variance. A flip that was the best available bet but revealed a strong enemy piece gets flagged as a "mistake" identical to a genuinely bad choice.

Key point: **both components are computable from non-god-view evals.** When the engine evaluates a pre-flip position it averages over all possible tiles (Star1 chance nodes) without knowing the actual one, so "expected value of flipping b3" is an honest number from public info.

## Worked example

Game `bq_c0b25fec-4ef6-4520-9790-a42a64eedbb8` (banqi, human red lost). Analysis flagged **5 inaccuracies, 0 mistakes, 0 blunders** — all on flips — yet red lost decisively. Eval series (red POV, cp): roughly even through ply 4 (+34), then a steady collapse to −755 by ply 13, grinding to a lost endgame. Two effects combined:

- **Floor compression:** once red is at ~20% win (ply 10), no single move can be a "blunder" — there isn't enough win% left to lose. A slow bleed reads as a string of inaccuracies.
- **Chance attribution:** each flagged flip's swing mixed the decision with the reveal. We can't currently tell which of red's flips were bad *choices* vs bad *luck*.

## The gap (smaller than "build MultiPV")

The hard primitive already exists:

- **MistyBanqi** has `root_move_values()` (`banqi_rust/src/engine.rs`) — every root move's exact value, flips resolved via Star1 expectation. It's marked diagnostic and **not exposed over UCI**.
- **MistyJungleFlip** computes per-move values in its root loop but has no collected accessor — a small add mirroring banqi.

Three layers:

1. **Engine:** surface per-root-move EVs over UCI (MultiPV or `go searchmoves <move>`). Banqi ≈ plumb the existing function; jungle-flip add the function first. Version bump + release + rebake ×2.
2. **Server analysis:** swap the per-ply single-PV query for one root-values query per ply (same search, keep the siblings); cache `{bestEV, playedEV}` per move.
3. **Client:** `decision loss = winPct(bestEV) − winPct(playedEV)` drives the glyph; `luck = winPct(realized) − winPct(playedEV)` is shown separately, ungraded.

This is essentially poker analysis for board games (grade the bet's EV, treat the outcome as variance).

## Proposed UI

- **Advantage graph — two lines + luck band.** Solid = realized (truth, includes luck). Ghost = "if flips ran average" (realized minus cumulative your-luck). The shaded gap between them *is* your variance; downward gap = flips hurt you.
- **Per move — a four-quadrant chip.** A skill glyph (✓/?!/?/??) AND a neutral 🎲 luck value. "right choice, unlucky" vs "wrong choice" at a glance.
- **Game summary — two scores.** Decision accuracy (skill; the only thing that should feed a rating) and Luck (net ±win%). e.g. "76% decisions, −19% on flips."
- **Gamble lens (flagship).** On a flip decision, overlay the bet using only public info: "45% yours (avg +0.3) / 55% enemy (avg −0.5), EV −0.1" + ghost arrows for best flip vs your flip. Board shows the bet; graph shows how it landed.

## Interim shipped (this issue's stopgap)

Flips are now left **unjudged** (no glyph, no "was best", not counted, excluded from ACPL) — the server marks flip plies as `chancePlies` and `computeGameAnalysis` skips them. This stops the active mislabeling ("5 inaccuracies") until the decomposition lands. Eval graph still shows the honest collapse.

## Sequencing

1. ✅ Interim: stop glyphing flips.
2. Engine: expose per-move EVs (banqi first — nearly free — as proof).
3. Decomposition end-to-end: two-line graph + luck chips first (highest insight/effort), gamble lens as the flagship follow-up.

## Related

- Engine strength / eval flatness: brianhliou/misty-jungle#2, #3 (distinct — those are eval *quality*; this is eval *attribution*).

**Comment by brianhliou** [comment]
## Update: jieqi is the lead proof variant (Pikafish gives honest chance EVs — source-confirmed)

Jieqi (reveal xiangqi) was missed in the surface work — same hidden-info/reveal-chance class as banqi/jungle-flip, **and its analysis surface is not built** (jungle/banqi/jungle-flip are). It should be built alongside them regardless.

The bigger finding: **PikaJieQi already computes the honest per-move expected values this issue needs.** Verified by reading the `jieqi_old` source (`~/projects/tools/pikafish-jieqi-old/src`):

- Dark pieces are modeled as **chance nodes**: `search.cpp` enumerates every possible reveal (`while (pos.getDark(...))`, bounded by `MAXDARKDEPTH 4`), weights each outcome by its unrevealed-pool count (`darkcount`), and combines via `ScoreCalc` (misc.h): `_totalScore += score * count; … CalcEvg() = _totalScore / _totalCount` — a **probability-weighted expectation** over the pool. No determinization, no god-view.
- Nuance: `CalcEvg` is **downside-adjusted**, not a pure mean — when the average is much rosier than the worst case (`evg − min > DARKMAXDIFF`), it returns the worst case. Fine (arguably good) for grading a *decision*, but the number isn't a straight average.
- Pikafish is Stockfish-family → **native MultiPV**. So per-move EVs are **zero engine work** (flip on `setoption MultiPV`), unlike our Rust engines (which need `root_move_values` exposed).

### Sharpened sequencing
1. **Prove the decomposition on jieqi first.** Only work is the analysis **surface** (server reconstruction from the redacted per-ply FEN + route + client), same shape as banqi. MultiPV then yields per-move EVs for free.
2. **Banqi second, as a comparison data point.** Its `root_move_values` is a *pure* mean; comparing banqi (raw EV) vs jieqi (downside-adjusted EV) tells us which framing reads better before committing the UI.
3. **Deprioritize** the custom-engine EV-exposure work (`root_move_values`→UCI for banqi/jungle-flip) — only needed to bring them to jieqi's level, decide after jieqi proves the UX.

### Build path for jieqi
- Layer 1 (needed regardless): basic jieqi analysis surface mirroring banqi-analysis.ts — reconstruct per-ply from the deal/reveal events, feed PikaJieQi the redacted FEN, cache, fail-closed, client underboard. Brings jieqi to parity with the other 3.
- Layer 2 (the decomposition): add MultiPV per-move EV capture → `{bestEV, playedEV, realized}` per move → decision-loss/luck math → UI (four-quadrant chips + two-number summary first, then two-line graph, then gamble lens).

Interim (flips left unjudged via `chancePlies`) already shipped for banqi/jungle-flip; extend the same to jieqi when its surface lands.

**Comment by brianhliou** [comment]
Shipped: banqi decomposition in d9e13f43 (PR #213), jungle-flip in 5e4e912a (PR #214), hidden-pool rebalance fix in 15a87416 (#221). Decision-vs-luck decomposition is live in prod for banqi and jungle-flip. Closing as completed.
